### DISCUSSION TITLE
How to ensure Micronaut app start even if database is down

### DISCUSSION THREAD

**Discussion by cliftonz1** [micronaut-projects/micronaut-data#1820] [Q&A]
We are wanting to ensure that even if the database is down on startup, micronut data does not fail and keeps looking to see if the data base is up instead of failing.

**Comment by graemerocher** [comment]
you could define your own named bean that implements `DataSource` and has some kind of retry behaviour

**Reply by dstepanov** [reply]
It might be possible with some JDBC pool configuration https://github.com/brettwooldridge/HikariCP

**Reply by graemerocher** [reply]
https://github.com/brettwooldridge/HikariCP/wiki/Bad-Behavior:-Handling-Database-Down looks useful
